mqtt/tls-connect-to-broker.md

```diff
@@ -0,0 +1,116 @@
+certfile /<path-to-LE-cert-files>/<fullchain>.cer
+keyfile /<path-to-LE-cert-files>/<private-key-name>.key
+## Forces use of modern version of TLS to avoid security issues
+tls_version tlsv1.2
+
+## Forces ALL CLIENTs to provide a valid certificate - change the node config to allow this from NR
+#require_certificate true
+~~~
+{: .shell}
+{{ page.lcb }}% endraw %}
+
+After making these changes, you have to restart the mosquitto broker using the command:
+
+~~~text
+[~]$ sudo systemctl restart mosquitto
+~~~
+{: .shell}
+
+**<code class="node">MQTT Config</code> node configuration**
+
+![](/images/mqtt/tls-connect-1.png)
+
+Notes
+
+* You need to use the IP name rather than IP address in the server name if using Let's Encrypt (otherwise the certificate isn't valid).
+* You need to change the server name to a url, prefixed with `mqtts://`. This disables the port field, I change that first to `8883` to remind me what the correct port will be.
+* You **do not** need to set the "Enable secure connection" flag. That lets you authenticate the Node-RED client connection to the broker (if you set the require_certificate to true for example).
+
+### Discussion
+
+Mosquitto allows you to create multiple ports for connectivity. This lets you use websockets and TLS encrypted connections in addition to the default connection.
+
+The folder `/etc/mosquitto/conf.d/` can contain any number of config files which will all be applied so that you can split your custom changes into separate files if you like.
+
+Just remember that once you use a custom file to set ports, the default port (1883) is no longer active so you have to specify that as well. The standard port for MQTT over TLS (MQTTS) is 8883. You can, however, use other ports if they are not in use. Make sure you use a port number greater than 1024 otherwise everything that wants to use that port has to have root privalages.
+
+You can check which ports the broker has opened with the command:
+
+~~~text
+[~]$ sudo netstat -lptu | grep mosquitto
+tcp        0      0 0.0.0.0:8883            0.0.0.0:*               LISTEN      17697/mosquitto
+tcp        0      0 0.0.0.0:1883            0.0.0.0:*               LISTEN      17697/mosquitto
+tcp6       0      0 [::]:8883               [::]:*                  LISTEN      17697/mosquitto
+tcp6       0      0 [::]:1883               [::]:*                  LISTEN      17697/mosquitto
+~~~
+{: .shell}
+
+You can test whether the server device is allowing connections on a port by using telnet from another device.
+
+~~~text
+[~]$ telnet <ip-name> 8883
+~~~
+{: .shell}
+
+If the connection opens, then the target device is accepting connections on that port.
+
+Note that the operating system automatically opens the required ports through the devices firewall.
+
+If you want to monitor what the broker is doing, including seeing which clients connect to which ports, use the following command:
+
+~~~text
+[~]$ sudo tail /var/log/mosquitto/mosquitto.log -f
```

Comment:
Implementation specific...

Reply:
clarified

Reply:
Am I missing something? All the commands in the discussion part still seem to be platform/installation specific without a hint...